### PR TITLE
Update go, remove deprecations and house-keeping

### PR DESCRIPTION
**Pull request by alexanderstephan**
- Update to 1.19, including the images used in Dockerfiles
- Staticcheck runs now for every (relevant) project in this repo
- Update staticcheck (looks like this is not detected by Dependabot, we missed out on a few versions)
- `io/ioutil` is deprecated (actually since 1.16), it can be replaced with functions from `os` and `io`
- Fix some other stuff found by staticcheck

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -4,17 +4,17 @@ WORKDIR /app
...
-RUN rm -rf web/assets/ts-dist
-RUN rm -rf web/assets/css-dist
+RUN rm -rf web/assets/ts-dist &&\
+    rm -rf web/assets/css-dist
 
 WORKDIR /app/web
 RUN npm i --no-dev
 
-FROM golang:1.19-alpine as build-env
+FROM golang:1.19.1-alpine3.16 as build-env
```
Comment (joschahenningsen) [inline Dockerfile]: I think golang:1.19 should automatically update until golang:1.20 is released.

File: Dockerfile [hunk 2]
```diff
@@ -28,7 +28,7 @@ COPY --from=node /app/web/node_modules ./web/node_modules
 ARG version=dev
 RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags "-w -extldflags '-static' -X main.VersionTag=${version}" -o /go/bin/tumlive cmd/tumlive/tumlive.go
 
-FROM alpine:3.16
+FROM alpine:3.16.2
```
Comment (joschahenningsen) [inline Dockerfile]: Same here, alpine:3.16 should always be the latest alpine:3.16.x

File: go.mod [hunk 3]
```diff
@@ -1,6 +1,6 @@
 module github.com/joschahenningsen/TUM-Live
 
-go 1.18
+go 1.19
```
Comment (joschahenningsen) [inline go.mod]: The version specified in go.mod is a constraint about the minimum go version required to run the app: https://go.dev/ref/mod#go-mod-file-go

In our case we could still allow running tum-live with go 1.18

File: worker/Dockerfile [hunk 4]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.19-alpine3.16 as builder
+FROM golang:1.19.1-alpine3.16 as builder
```
Comment (joschahenningsen) [inline worker/Dockerfile]: golang:1.19-alpine3.16 should be fine like above

File: worker/Dockerfile [hunk 5]
```diff
@@ -13,20 +13,17 @@ RUN GO111MODULE=on go mod download && go mod verify
 ARG version=dev
 RUN GOOS=linux GOARCH=amd64 go build -a -ldflags "-w -X main.VersionTag=${version}" -o /worker cmd/worker/worker.go
 
-FROM alpine:3.16
+FROM alpine:3.16.2
```
Comment (joschahenningsen) [inline worker/Dockerfile]: here as well

File: worker/Dockerfile [hunk 6]
```diff
@@ -13,20 +13,17 @@ RUN GO111MODULE=on go mod download && go mod verify
...
-COPY --from=builder /usr/lib/libtesseract*.so.* /usr/lib/
-COPY --from=builder /usr/lib/libgif*.so.* /usr/lib/
-COPY --from=builder /usr/lib/libtiff*.so.* /usr/lib/
-COPY --from=builder /usr/lib/libzstd*.so.* /usr/lib/
-COPY --from=builder /usr/lib/liblzma*.so.* /usr/lib/
-
+COPY --from=builder \
+    /usr/lib/liblept*.so.* /usr/lib/libtesseract*.so.* \
+    /usr/lib/libgif*.so.*  /usr/lib/libtiff*.so.* \
+    /usr/lib/libzstd*.so.* /usr/lib/liblzma*.so.* /usr/lib/
```
Comment (joschahenningsen) [inline worker/Dockerfile:26]: Nice!

File: worker/edge/Dockerfile [hunk 7]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.19-alpine3.15 as builder
+FROM golang:1.19.1-alpine3.16 as builder
```
Comment (joschahenningsen) [inline worker/edge/Dockerfile]:
```suggestion
FROM golang:1.19-alpine3.16 as builder
```

**Review by joschahenningsen — COMMENTED** [review]
Very nice! Review suggestions are nits :)

**Review by joschahenningsen — APPROVED** [review]
Thanks!

**Comment by alexanderstephan** [comment]
Now the CI fails. 😢

**Comment by joschahenningsen** [comment]
Yeah, that's on me though, I'll fix it :D

**Pull request merged**